crates/turborepo-lib/src/engine/builder.rs

```diff
@@ -389,6 +393,21 @@ impl Error {
     }
 }
 
+const INVALID_TOKENS: &[&str] = &["$colon$", "/", "\\"];
```

Comment:
Do we need to ban `/` and `\\`? If I understand correctly, we only have weird behavior because of these for the log files. If we escaped these as well in the log file paths, then we don't need to ban them. (I'm afraid we're going to find a lot of projects use `/` in script names)

Reply:
Ugh. Ok. I'll limit it to just `$colon$` for now, but leave the mechanism in place to add more.

I kinda suspect cross-platform caching might be broken if you use those tokens, but that isn't necessarily a reason to ban them, just that they aren't cross-platform-safe